llama_index/chat_engine/types.py

```diff
@@ -155,7 +187,7 @@ def stream_chat(
     @abstractmethod
     async def achat(
         self, message: str, chat_history: Optional[List[ChatMessage]] = None
-    ) -> AgentChatResponse:
+    ) -> Union[AgentChatResponse, StreamingAgentChatResponse]:
```

Comment:
If you wanted to simplify this, you can define `AGENT_RESPONSE_TYPE = Union[...]` in the types file

We do something similar for the query engine `Response`/`StreamingResponse`

Reply:
Done!